### ISSUE TITLE
Allow building shared libraries

### ISSUE THREAD

**Issue by tomfi** [mike-neck/graalvm-native-image-plugin#145]
Is it possible to add the option / flag to build a shared library with this plugin?

When adding the `--shared` flag to GraalVM it should not require a `main-class` anymore (https://www.graalvm.org/reference-manual/native-image/ImplementingNativeMethodsInJavaWithSVM/#create-a-shared-library)

At the moment the maven plugin support it quite well but i could not find a good gradle plugin that supports it.

As a workaround i did the following configuration:
```
nativeImage {
  mainClass = "demo.MainKt"
  executableName = project.name
  arguments(
    "--shared",
  )
}
```

Where `demo.MainKt` is a stub:
```
package demo

fun main() {
}

```

This results in building a shared library by graalvm as expected.

It would be great if the `mainClass` and `executableName` were optional to allow building shared libraries with this plugin without workarounds.

Thanks in advance, this plugin is great 👍

**Comment by mike-neck** [comment]
The `mainClass` property cannot be optional, because `nativeImage` task and `generateNativeImageConfig` task are sharing the same `mainClass` property. Without setting `mainClass` property, the later task(`generateNativeImageConfig`) will fail.
This means that we needs refactoring this plugin for realizing this feature. We have to introduce a convention object with the same interface to `nativeImage` task.

**Comment by mike-neck** [comment]
The shared library feature is limited to Java11.

**Comment by mike-neck** [comment]
In PR #147 

- deprecate `mainClass` property in `nativeImage` task
- add `buildType {}` block to configure the type of build(executable/shared library).


Example configuration of building shared library

```groovy
nativeImage {
  buildType { BuildTypeSelector build ->
    build.sharedLibrary
  }
}
```

Example configuration of building executable

```groovy
nativeImage {
  buildType { BuildTypeSelector build ->
    build.executable { main = 'com.example.Main' }
  }
}
```

**Comment by mike-neck** [comment]
The feature will be available at `v1.4.0`.

**Comment by tomfi** [comment]
@mike-neck thank you very much :)
